test/common.py

```diff
@@ -1359,7 +1362,7 @@ def get_library(self, name, generated_libs, configure=['sh', './configure'],  #
     # try to pass linker settings when compiling).
     emcc_args = []
     if not native:
-      emcc_args = self.get_emcc_args(ldflags=False)
+      emcc_args = self.get_emcc_args(compile_only=False)
```

Comment:
I am confused: This line maps `compile_only=False` to `ldflags=False`, while all other places map them to opposite booleans. The opposite is what I would expect ("compile only" means *not* to use link-time flags") so maybe this is in error, or I am misreading it?

Reply:
Ah yes, I add some issue with `test_poppler` where it seems it was depending on `-sERROR_ON_UNDEFINED_SYMBOLS=0` being the cflags (which is a linker flag).   I'm digging deeper now.

Reply:
OK, fixed by modifying `get_poppler_library` to avoid `set_setting`.